### PR TITLE
Add github dark

### PR DESCRIPTION
**Pull request by steff456**
This PR is part of https://github.com/Quansight-Labs/jupyterlab-accessible-themes/issues/40

For adding a new theme a new directory is created with the following structure,

```
├── theme_name
│   ├── src
│   │   ├── index.ts
│   ├── style
│   │   ├── index.css
│   │   ├── variables.css
│   ├── package.json
│   ├── tsconfig.json
│   ├── README.md
```

A JupyterLab instance will start in your browser and you can select the theme, by clicking `Settings > Theme > Github Dark` in the main menu.

Some screenshots on how it looks right now,

image

New theme checklist:

- [x]  New directory with new package
- [x] Update the setup.py to include the new package
- [x] Create a README.md inside the new package with information about the color palette and resources
- [x] Update main README.md with new theme

### CODE REVIEW

**Comment by trallard** [comment]
@steff456 can you rebase against main? or merge main? there are merge conflicts here

**Review by gabalafou — CHANGES_REQUESTED** [review]
Yay! I was able to install and load this theme! I like it!

My one concern is that to my eyes it looks like a lot of contrast has been "washed out" with this theme. Here's a screenshot of the launcher, for example:



I can barely tell where the background ends and any given launcher card begins. And there are a few other places that appear low contrast to my eyes.

Is this something we need to fix now or later?

I'm marking this "request changes" because even if now is not the time to modify this GitHub Dark theme, the question about when or if it needs to be fixed needs to be resolved before I feel good about hitting the approve button. Thanks!

File: README.md
Comment (gabalafou) [inline README.md:43]:
```suggestion
- [GitHub Light](./packages/githublight/README.md) - Color-blind friendly
```

or `GitHub dark`, but we should be consistent, either way

**Comment by trallard** [comment]
> Is this something we need to fix now or later?

This should be fixed now rather than later cc/ @steff456

**Comment by steff456** [comment]
I updated the color of all the borders, just for them to be more visible. Now the interface looks like this,

image

**Comment by trallard** [comment]
In the README and the conda-forge feedstock, we are requesting Python >= 3.6. See #56 3.7 has already reached EOL so we should be requesting 3.8 or above really

**Comment by trallard** [comment]
Noticed again that the schema error pops with this theme too (see #57)

There are also several issues I have identified in #59 that need fixing

File: README.md
Comment (trallard) [inline README.md:44]: Is this so? There are both GH dark and GH dark colourblind friendly https://github.com/primer/github-vscode-theme if the latter then it seems the name of the theme should be corrected

**Review by trallard — CHANGES_REQUESTED** [review]

**Comment by steff456** [comment]
> In the README and the conda-forge feedstock, we are requesting Python >= 3.6. See https://github.com/Quansight-Labs/jupyterlab-accessible-themes/issues/56 3.7 has already reached EOL so we should be requesting 3.8 or above really

I saw this item in the tracking issue for the next release and will make a separate PR fixing it.

File: README.md
Comment (steff456) [reply README.md:44]: In the VSCode documentation it states that all the themes support different types of color blindness, it is not specific to which one though 🤷‍♀️

Do you prefer to only add the `color-blind friendly` note to the themes that are named specifically?

Comment (trallard) [reply README.md:44]: Unless a theme is guaranteed to be colourblind-friendly, we should not say it is (otherwise, folks will expect this to be the case).

From the VSCode extension, https://github.com/primer/github-vscode-theme#install, some themes are colourblind friendly - if in this PR you *are not* implementing any of those marked as colourblind friendly, then this should not be called out. (my understanding was you were adding the light and dark default ones)

> GitHub Light Default
GitHub Light High Contrast ✨ new ✨
GitHub Light Colorblind ✨ new ✨
GitHub Dark Default
GitHub Dark High Contrast
GitHub Dark Colorblind ✨ new ✨
GitHub Dark Dimmed

Comment (steff456) [reply README.md:44]: ok! I'll make the change

**Comment by steff456** [comment]
@trallard the last commits fixed the issues for this theme of both https://github.com/Quansight-Labs/jupyterlab-accessible-themes/issues/59 and https://github.com/Quansight-Labs/jupyterlab-accessible-themes/issues/57. Please check it and let me know if anything else is needed to merge this PR 🙃

**Comment by trallard** [comment]
Seems the schema error was not fixed here see:

<details>
```bash
[W 2023-07-07 14:10:04.166 LabApp] wrote error: 'Schema not found: /Users/tania/mambaforge/envs/jlab-themes/share/jupyter/lab/schemas/jupyterlab-accessible-themes/pitayasmoothie.json'
    Traceback (most recent call last):
      File "/Users/tania/mambaforge/envs/jlab-themes/lib/python3.10/site-packages/tornado/web.py", line 1784, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/Users/tania/mambaforge/envs/jlab-themes/lib/python3.10/site-packages/tornado/web.py", line 3290, in wrapper
        return method(self, *args, **kwargs)
      File "/Users/tania/mambaforge/envs/jlab-themes/lib/python3.10/site-packages/jupyterlab_server/settings_handler.py", line 48, in get
        result, warnings = get_settings(
      File "/Users/tania/mambaforge/envs/jlab-themes/lib/python3.10/site-packages/jupyterlab_server/settings_utils.py", line 370, in get_settings
        schema, version = _get_schema(schemas_dir, schema_name, overrides, labextensions_path)
      File "/Users/tania/mambaforge/envs/jlab-themes/lib/python3.10/site-packages/jupyterlab_server/settings_utils.py", line 47, in _get_schema
        raise web.HTTPError(404, notfound_error % path)
    tornado.web.HTTPError: HTTP 404: Not Found (Schema not found: /Users/tania/mambaforge/envs/jlab-themes/share/jupyter/lab/schemas/jupyterlab-accessible-themes/pitayasmoothie.json)
[W 2023-07-07 14:10:04.167 LabApp] 404 GET /lab/api/settings/jupyterlab-accessible-themes:pitayasmoothie?1688735404102 (e23722994c7b43e983d7df74432af902@::1) 17.01ms referer=http://localhost:8888/lab
```
</details>

but there is another PR named "Fix schema error" #61 - can you confirm which addresses the schema error in #57?

**Comment by trallard** [comment]
Some issues that have yet to be fixed:

- [ ] Selected file colours are still not WCAG compliant - there should be a 3:1 contrast between `.jp-DirListing-item.jp-mod-selected` and the background and 4.5:1 between text and `.jp-DirListing-item.jp-mod-selected`

Colour_Contrast_Analyser__CCA_

- [ ] Same issue on hover on menu items hover on background fails to meet 3:1

Colour_Contrast_Analyser__CCA__and_lab_-_JupyterLab

- [ ] dividers are not noticeable

lab_-_JupyterLab

- [ ] Icons not having enough contrast against background

lab_-_JupyterLab

Colour_Contrast_Analyser__CCA__and_JupyterLab

- [ ] For whatever reason the dividers in the menus are hoverable when they are not in the default JLab themes, this seems to be a bug


Untitled_ipynb_-_JupyterLab
Untitled_ipynb_-_JupyterLab


- [ ] New tab button does not have enough contrast against background 
Colour_Contrast_Analyser__CCA__and_Untitled_ipynb_-_JupyterLab

@steff456 I suggest you check the colours with a contrast checker or a web extension to ensure the colours meet WCAG AA standards

**Comment by steff456** [comment]
@trallard The last commit addresses all the issues you found. 

I tried the suggestion @gabalafou gave me last week of trying lighthouse and axe chrome extensions to test for colors, but the tools didn't flag any color contrast issues. 

Please play around with the theme and let me know if there's something else that's off. I'm testing as much as I can on my end but there's a lot of parts that I may be missing.

**Review by gabalafou — CHANGES_REQUESTED** [review]
@steff456 I see what you mean about Axe Chrome Dev Tools not catching contrast issues.

Here's a screenshot of the toast notification asking the user if they would like to receive JupyterLab news. With this GitHub Dark theme applied, the background color of the buttons (open privacy policy, yes, no) is indistinguishable from the toast background, so they don't even look like buttons: 



But Axe didn't catch that. 

It also didn't catch what I bet is a problematic contrast issue between the yaml file icon and the file browser background:

filebrowser screenshot showing poorly contrasting yaml file

Nor did it catch the lack of contrast between the menubar color and the active menu item color. 

Another thing—not contrast-related—is that some of the vertical and horizontal divider lines in the UI have been messed up. In following screenshot, for example, if you look closely in the middle you can see a line that sticks out 1 to 2 pixels too far:



In the other themes, I think the border color matches the background in certain spots, so you don't see these little visual bugs.

I also can't seem to open the property inspector or the debugger from the right side bar. Are you seeing the same?

**Comment by gabalafou** [comment]
Figured out the issue with the property inspector and the debugger. With GitHub Dark theme, they don't have minimum widths, so I can drag them to zero width.

**Comment by steff456** [comment]
The latest changes fix the issues you flagged @gabalafou 

- [x] Notification center buttons are now visible

image

image

- [x] Contrast issue between some icons in the files menu and the background

image

- [x] Add a minimum sidebar width value

**Comment by gabalafou** [comment]
Not sure if the following problems should block merging this PR but I'm noting them down so they're recorded somewhere.

- [ ] (1) Not enough contrast between the blue `button:focus-visible` focus ring and the dark background: image

Color Contrast Analyzer app shows that outline blue of #0D47A1 against black #212121 has 1.9:1 contrast ratio which fails every WCAG 2.1 contrast requirement

Side note: JupyterLab Dark has the same low contrast problem with the focus ring.

- [ ] (2) No visible focus indicator for buttons in toast notification:



On inspection the issue seems to be the following CSS rule:

```css
.jp-toast-button:focus {
    outline: 1px solid var(--jp-reject-color-normal, var(--jp-layout-color2));
}
```

Dev Tools says that `--jp-reject-color-normal` is not defined, whereas the fallback `--jp-layout-color2` is set to `--gh-dark-gray8: #161b22` which is the same color that toast background is set to: 

```css
background: var(--toastify-color-light);
--toastify-color-light: var(--jp-layout-color1);
--jp-layout-color1: var(--gh-dark-gray8);
--gh-dark-gray8: #161b22;
```

Not sure if the issue here is that --jp-layout-color2 and --jp-layout-color-1 are the same, or if the issue is `--jp-reject-color-normal` not being defined, or the `Toastify__toast-theme--light` class getting applied when this is a dark theme, or something else.

- [ ] (3) When hovered, the "Do Not Check For Updates" button in the toast notification seems to have a transparent background, so it ends up looking like text instead of a button when hovered:



- [ ] (4) Not enough contrast for green status dot for notebook file in file browser: 

Color Contrast App showing that green #196C2E against black #161B22 has contrast ratio 2.7:1 and meets no WCAG 2.1 contrast requirements

- [ ] (5) Not enough contrast for package names in License viewer:  localhost_8888_lab_workspaces_auto-V_tree_jupyterlab_accessible_themes

image

I'm going to stop there. There are probably other things I didn't catch, but overall, at time of this review, the theme seems in pretty good shape to me.

**Comment by trallard** [comment]
ok to get this PR in, I would say let's fix the following (they seem scoped and small enough)

- [ ] (4) Not enough contrast for green status dot for notebook file in the file browser
- [ ] (3) When hovered, the "Do Not Check For Updates" button in the toast notification seems to have a transparent background, so it ends up looking like text instead of a button when hovered

The other items: @steff456  can you move these to a new issue so we can keep track just in case?
Once those are sorted we will merge

**Comment by steff456** [comment]
The last commit fixes all the issues raised by @gabalafou,

1. Now the focus indicator has the same blue as in other places,

image

image

2. Visible focus indicator for buttons in toast notifications, (I left the same combination as above)

image

3. Fixed the hover state of the buttons in the toast notifications, now it will use a lighter shade of gray

image

4. Fixed the contrast for the green status for the notebook icon in the file browser,

image

image

5. Fixed the contrast for package names in License viewer

image

image

**Comment by trallard** [comment]
Thanks @steff456 this looks good to me. 
Will wait for @gabalafou to confirm but I think this is in a good shape to merge now.

**Review by trallard — APPROVED** [review]

**Review by gabalafou — APPROVED** [review]
Sorry I didn't realize this was waiting on me to push the button! :D

**Comment by trallard** [comment]
Thanks folks. @steff456 this is ready to merge now, thanks for seeing this lengthy contribution through 🙏🏽

**Comment by steff456** [comment]
Thanks!! I'm super happy to see this finally merged

**Pull request merged**
